NewForce-at-Mountwest/bangazon-workforce-gnocci
# HR Can View All Departments, Create Department(s), View Department Details (with Itemized Teammates)

# Description
Given the user wants to view departments, when the user 'clicks' on the "Departments" section in the navigation bar, then all current departments should be listed.  The following information should be presented to the user: Department Name, Department Budget, Size of Department (Number of Employees Assigned).  Given the user is viewing all departments, when the user 'clicks' on the "Create New Department" link, then a form should be presented in which the new department name can be entered.  Given the user is viewing list of departments, when the user clicks on a department, then a view should be presented with the department name as a header and a list of employees currently assigned to that department should be listed.  Given the user wants to see all of the employees in a department, when the user performs a gesture on the department detail affordance, then the user should see the department name, the department budget, and the full name of each employee in that [given] department.


## Related Ticket(s)
1. HR should be able to view all departments.
1. HR should be able to create a department.
1. HR should be able to view a department's details.
1. Department details should itemize teammates.


## Steps to Test
1. git add and git commit whatever you're working on (on your branch)
2. git checkout master, to switch to our local master branch
3. git fetch --all will give you access to branch in question
4. git checkout "jw-hr-create-dept" to switch to branch in question
5. If you don't see the most recent version of files, try git pull origin "jw-hr-create-dept"
1. in gitbash cd to bangazon-workforce-gnocci
1. start BangazonWorkforce.sln
1. add snippet to database
````````````
https://raw.githubusercontent.com/NewForce-at-Mountwest/bangazon-workforce-gnocci/master/DB_SEED-SQL.md
````````````
1. press F5 to run browser / CTRL+R, A to run tests
1. in browser, 'click' "Department" to reach Department Index
Department Index should show Department Name(s), Total Employees [in Department], Dept. Budget, and have the option to Delete the Department(s)* - - added feature
1. click "Create New Department"
1. user should see: "Name" and "Budget" form fields, with a "Create" Button - as well as link "Back to List of Departments"
1. populate Create Department form fields and hit "Create" Button OR 'click "Back to List of Departments" link
1. in Department Index, user can 'click' on Department Name (e.g. "Beekeeping")
1. if Department Name is 'clicked', Department Details should be shown
1. Department Details include Dept. Budget and all [Employee] Team Members (if No Employees are in Department, it will list "No Employees")


# Checklist:
